File: ASTICLibrary/src/astic/dpm/FileType.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package astic.dpm;

import java.io.File;

/**
 *
 * @author Elvis
 */
public enum FileType {
    App,Video,Audio,Photo,Text,Other;
    
    public static FileType fromExtension(String extension)
    {
        String [] mAppExt = {"exe"};
        String [] mVideo = {"mp4","avi"};
        String [] mAudio = {"mp3","wav"};
        String [] mPhoto = {"jpg","png"};
        String [] mText = {"txt"};
        for(String ext:mAppExt)
            if(ext.equals(extension))
                return App;
        for(String ext:mVideo)
            if(ext.equals(extension))
                return Video;
        for(String ext:mAudio)
            if(ext.equals(extension))
                return Audio;
        for(String ext:mPhoto)
            if(ext.equals(extension))
                return Photo;
        for(String ext:mText)
            if(ext.equals(extension))
                return Text;
        return Other;
    } 
    
    public static FileType fromFile(File file)
    {
        String name = file.getName();
        int index = name.lastIndexOf('.');
        String ext = null;
        if(index>0)
            ext = name.substring(index+1);
        //System.out.println(""+ext);
        if(ext!=null&&ext.length()>0)
            return fromExtension(ext);
        else
            return FileType.Other;
    }
  
   
}

File: ASTIC_client_V2/ASTIC_client_V2/model/FileType.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Windows;
using System.Windows.Controls;
using System.Windows.Data;
using System.Windows.Documents;
using System.Windows.Input;
using System.Windows.Media;
using System.Windows.Media.Imaging;
using System.Windows.Shapes;

namespace ASTIC_client_V2
{
	public enum FileType
	{
		App,
        Video,
        Audio,
        Photo,
        Text,
         Other,
        All
        
	}

    public class FileTypeFactory
    {
        public static FileType FromFile(String path)
        {
            String fileName = System.IO.Path.GetFileName(path);
            if(fileName.Contains("."))
            {
                return FromExt(fileName.Substring(fileName.LastIndexOf('.')));
            }else{
                return FileType.Other;
            }
        }

        private static FileType FromExt(String ext)
        {
            switch (ext)
            {
                case ".txt" :
                case ".pdf" :
                    return FileType.Text;
                case ".jpg":
                case ".png":
                    return FileType.Photo;
                case ".mp4":
                case ".avi":
                case ".mkv":
                    return FileType.Video;
                case ".mp3":
                case ".wav":
                    return FileType.Audio;
                case ".exe":
                    return FileType.App;
            }
            return FileType.Other;
        }
    }
}